application/models/Iota.php

```diff
@@ -135,34 +128,41 @@ function fetchIota($postdata) {
 			$sql .= " and exists (select 1 from " . $this->config->item('table_name') . " where station_id in (". $location_list . ") and col_iota = iota.tag";
 
 			if ($postdata['mode'] != 'All') {
-				$sql .= " and (col_mode = '" . $postdata['mode'] . "' or col_submode = '" . $postdata['mode'] . "')";
+				$sql .= " and (col_mode = ? or col_submode = ?)";
+				$binding[] = $postdata['mode'];
+				$binding[] = $postdata['mode'];
 			}
 
 			if ($postdata['band'] != 'All') {
 				if ($postdata['band'] == 'SAT') {
-					$sql .= " and col_prop_mode ='" . $postdata['band'] . "'";
+					$sql .= " and col_prop_mode = ?";
+					$binding[] = $postdata['band'];
 				} else {
 					$sql .= " and col_prop_mode !='SAT'";
-					$sql .= " and col_band ='" . $postdata['band'] . "'";
+					$sql .= " and col_band = ?";
+					$binding[] = $postdata['band'];
 				}
 			}
 			$sql .= ")";
 		}
 
 		$sql .= ' order by tag';
-		$query = $this->db->query($sql);
+		$query = $this->db->query($sql, $binding);
 
 		return $query->result();
 	}
 
 	function getIotaWorked($location_list, $postdata) {
+		$binding = [];
+
 		$sql = "SELECT distinct UPPER(col_iota) as tag FROM " . $this->config->item('table_name') . " thcv
 			join iota on thcv.col_iota = iota.tag
 			where station_id in (" . $location_list .  ") and thcv.col_iota is not null
 			and not exists (select 1 from ". $this->config->item('table_name') . " where station_id = ". $location_list .  " and col_iota = thcv.col_iota)";
 
 		if ($postdata['mode'] != 'All') {
-			$sql .= " and (col_mode = '" . $postdata['mode'] . "' or col_submode = '" . $postdata['mode'] . "')";
+			$sql .= " and (col_mode = ? or col_submode = ?)";
+			$binding[] = $postdata['mode'];
```

Comment:
You have to add `$binding[] = $postdata['mode'];` twice because you need two binds here.

```suggestion
			$binding[] = $postdata['mode'];
            $binding[] = $postdata['mode'];
```